oe_theme.theme

```diff
@@ -714,3 +714,58 @@ function oe_theme_preprocess_pattern_list_item__variant_date__preview(array &$va
 function oe_theme_preprocess_pattern_date_block__preview(array &$variables): void {
   $variables['date'] = DateValueObject::fromArray($variables['date']);
 }
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function oe_theme_theme_suggestions_form_alter(array &$suggestions, array $variables): void {
+  $suggestions[] = $variables['theme_hook_original'] . '__' . $variables['element']['#form_id'];
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_input__search(array &$variables): void {
+  _oe_theme_preprocess_input_text($variables['element']);
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_form__search_form(array &$variables): void {
+  _oe_theme_preprocess_input_text($variables['element']['keys']);
+  $variables['input'] = $variables['element']['keys']['ecl_array'];
+}
+
+/**
+ * Helper function to additionally preprocess textfield, search elements.
+ *
+ * Moves the label element to the input template and creates some helper
+ * variables to be used with the ECL templates.
+ *
+ * @param array $element
+ *   Set of element data.
+ */
+function _oe_theme_preprocess_input_text(array &$element): void {
```

Comment:
Helpers should be in another file maybe?

Reply:
I will create follow up ticket. Because this "helper" consistent to other analogical helper function inside oe_theme.theme file. I believe scope of this new ticket should be refining approach how we could decouple all future helper function (including files to oe_theme.theme (old scool approach), use classes or even using new twig filters).

Reply:
I think this should be done asap in order to keep a proper order.
the file oe_theme.theme should only contain the hooks, all other methods and functions should be in another file.